NumLib/ODESolver/NonlinearSolver.cpp

```diff
@@ -215,63 +215,65 @@ bool NonlinearSolver<NonlinearSolverTag::Newton>::solve(
         sys.applyKnownSolutionsNewton(J, res, minus_delta_x);
         INFO("[time] Applying Dirichlet BCs took %g s.", time_dirichlet.elapsed());
 
-        if (!sys.isLinear() && _convergence_criterion->hasResidualCheck())
+        if (_convergence_criterion->hasResidualCheck())
             _convergence_criterion->checkResidual(res);
 
-        BaseLib::RunTime time_linear_solver;
-        time_linear_solver.start();
-        bool iteration_succeeded = _linear_solver.solve(J, res, minus_delta_x);
-        INFO("[time] Linear solver took %g s.", time_linear_solver.elapsed());
-
-        if (!iteration_succeeded)
+        GlobalVector* x_old = nullptr;
+        if (_convergence_criterion->hasResidualCheck() && _convergence_criterion->isSatisfied())
```

Comment:
Actually it would be nice if one could just write `if (_convergence_criterion->isSatisfied())`. But there's an interplay between the convergence criterion classes and the nonlinear solver. So this is probably hard to obtain.